api/src/main/java/io/minio/credentials/AnonymousCredentialsProvider.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * MinIO Java SDK for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
```

Comment:
Anonymous does not have any credentials. You could remove all anonymous credential handling.